src/runtime/qurt_hvx.cpp

```diff
@@ -52,4 +52,129 @@ WEAK void halide_qurt_hvx_unlock_as_destructor(void *user_context, void * /*obj*
     halide_qurt_hvx_unlock(user_context);
 }
 
+/////////////////////////////////////////////////////////////////////////////
+//
+// halide_hexagon_prefetch_buffer_t
+//
+// Prefetch a multi-dimensional box (subset) from a larger array
+//
+//   dim_count  number of dimensions in array
+//   elem_size  size in bytes of one element
+//   num_elem   total number of elements in array
+//   buf        buffer_t describing box to prefetch
+//              (note: extent/stride/min are all assumed to be positive)
+//
+// Notes:
+//  - Prefetches can be queued up to 3 deep (MAX_PREFETCH)
+//  - If 3 are already pending, the oldest request is dropped
+//  - USR:PFA status bit is set to indicate that prefetches are in progress
+//  - A l2fetch with any subfield set to zero cancels all pending prefetches
+//  - The l2fetch starting address must be in mapped memory but the range
+//    prefetched can go into unmapped memory without raising an exception
+//
+// TODO: Opt: Generate more control code for prefetch_buffer_t directly in
+// TODO       Prefetch.cpp to avoid passing box info through a buffer_t
+// TODO       (which results in ~30 additional stores/loads)
+//
+#define MAX_PREFETCH    3
+#define MASK16          0xFFFF
+#define MIN(x,y)        (((x)<(y)) ? (x) : (y))
+#define MAX(x,y)        (((x)>(y)) ? (x) : (y))
+//
+WEAK int halide_hexagon_prefetch_buffer_t(const uint32_t dim_count, const int32_t elem_size,
```

Comment:
dim_count can be computed by iterating across the buffer_t extents until a zero extent is encountered...but since it was already known in the prefetch pass (dimension of the box) when I was generating the prefetch call I figured it would be better to pass it in so it wouldn't have to be recomputed.